libs/ui/src/themes/make_theme.ts

```diff
@@ -34,14 +64,123 @@ const colorThemes: Record<ColorMode, ColorTheme> = {
   legacy: COLOR_THEME_LEGACY,
 };
 
-// TODO: Actually implement these themes:
+const INCHES_PER_MM = 1 / 25.4;
+const PIXELS_PER_INCH = 72;
+
+function mmToPx(mm: number): number {
+  // TODO: This is just a stop-gap proxy - need to add device detection logic to
+  // determine what screen we're on (ELO/VSAP/laptop) and use pre-defined
+  // ratios, since we can measure physical display dimensions more accurately.
+  const { devicePixelRatio } = window;
+
+  return mm * INCHES_PER_MM * devicePixelRatio * PIXELS_PER_INCH;
+}
```

Comment:
So my understanding is that the `mmToPx` logic is consistent and correct across devices, just being math, and the only `TODO` is to determine the `devicePixelRatio` on each device rather than relying on the `window`'s value? Is the `window` value typically not correct?

Reply:
That's right - for the most part, the `window.devicePixelRatio` value seemed to be accurate on my browser (it's `2` on my macbook screen and `1` when I drag the window to my external non-retina monitor), but I wasn't seeing the same thing in kiosk-browser (stayed at `1` regardless of which screen I was using). So, that didn't give me too much confidence in using that value and I want to do a bit more testing on the ELO devices